packages/cypress/src/generators/configuration/configuration.ts

```diff
@@ -52,11 +54,19 @@ export async function configurationGenerator(
   if (!installedCypressVersion()) {
     tasks.push(await cypressInitGenerator(tree, opts));
   }
+  const nxJson = readNxJson(tree);
+  const hasPlugin = nxJson.plugins?.some((p) =>
+    typeof p === 'string'
+      ? p === '@nx/cypress/plugin'
+      : p.plugin === '@nx/cypress/plugin'
+  );
   if (opts.bundler === 'vite') {
     tasks.push(addDependenciesToPackageJson(tree, {}, { vite: viteVersion }));
   }
-  await addFiles(tree, opts);
-  addTarget(tree, opts);
+  await addFiles(tree, opts, hasPlugin);
+  if (!hasPlugin) {
```

Comment:
Should we mark this for deprecation in Nx 18 or 19?

Reply:
Hm.. probably? But not yet.. let's see how it goes first.